src/segment_plastic.go

```diff
@@ -0,0 +1,245 @@
+		return
+	}
+	for _, line := range output[1:] {
+		if len(line) < 3 {
+			continue
+		}
+		if matchString(`(?i)merge\s+from\s+[0-9]+\s*$`, line) {
+			g.Unmerged++
+			continue
+		}
+		code := line[0:2]
+		switch code {
+		case "LD":
+			g.Deleted++
+		case "AD", "PR":
+			g.Added++
+		case "LM":
+			g.Moved++
+		case "CH":
+			g.Modified++
+		}
+	}
+	g.Changed = g.Added > 0 || g.Deleted > 0 || g.Modified > 0 || g.Moved > 0 || g.Unmerged > 0
+}
+
+func (g *plastic) parseStatusChangeset(status string) int {
+	var csRegex = `STATUS\s+(?P<cs>[0-9]+?)\s`
+	cs, _ := strconv.Atoi(findNamedRegexMatch(csRegex, status)["cs"])
+	return cs
+}
+
+func (g *plastic) getHeadChangeset() int {
+	output := g.getCmCommandOutput("status", "--head", "--machinereadable")
+	var csRegex = `\bcs:(?P<cs>[0-9]+?)\s`
+	cs, _ := strconv.Atoi(findNamedRegexMatch(csRegex, output)["cs"])
+	return cs
+}
+
+func (g *plastic) getPlasticFileContents(folder, file string) string {
+	return strings.Trim(g.env.getFileContent(folder+"/"+file), " \r\n")
+}
+
+func (g *plastic) getSelector() string {
+	var ref string
+	selector := g.getPlasticFileContents(g.plasticWorkspaceFolder+"/.plastic", "plastic.selector")
+	// changeset
+	ref = g.parseChangesetSelector(selector)
+	if ref != "" {
+		return fmt.Sprintf("%s%s", g.props.getString(CommitIcon, "\uF417"), ref)
+	}
+	// fallback to label
+	ref = g.parseLabelSelector(selector)
+	if ref != "" {
+		return fmt.Sprintf("%s%s", g.props.getString(TagIcon, "\uF412"), ref)
+	}
+	// fallback to branch/smartbranch
+	ref = g.parseBranchSelector(selector)
+	if ref != "" {
+		ref = g.truncateBranch(ref)
+	}
+	return fmt.Sprintf("%s%s", g.props.getString(BranchIcon, "\uE0A0"), ref)
+}
+
+func (g *plastic) parseChangesetSelector(selector string) string {
+	var csRegex = `\bchangeset "(?P<cs>[0-9]+?)"`
+	return findNamedRegexMatch(csRegex, selector)["cs"]
+}
+
+func (g *plastic) parseLabelSelector(selector string) string {
+	var labelRegex = `label "(?P<label>[a-zA-Z0-9\-\_]+?)"`
+	return findNamedRegexMatch(labelRegex, selector)["label"]
+}
+
+func (g *plastic) parseBranchSelector(selector string) string {
+	var branchRegex = `branch "(?P<branch>[\/a-zA-Z0-9\-\_]+?)"`
+	return findNamedRegexMatch(branchRegex, selector)["branch"]
+}
+
+func (g *plastic) init(props properties, env environmentInfo) {
+	g.props = props
+	g.env = env
+}
+
+func (g *plastic) truncateBranch(branch string) string {
+	fullBranchPath := g.props.getBool(FullBranchPath, false)
+	maxLength := g.props.getInt(BranchMaxLength, 0)
+	if !fullBranchPath && len(branch) > 0 {
+		index := strings.LastIndex(branch, "/")
+		branch = branch[index+1:]
+	}
+	if maxLength == 0 || len(branch) <= maxLength {
+		return branch
+	}
+	symbol := g.props.getString(TruncateSymbol, "")
+	return branch[0:maxLength] + symbol
+}
+
+func (g *plastic) getCmCommandOutput(args ...string) string {
+	val, _ := g.env.runCommand(g.getCmCommand(), args...)
+	return val
+}
+
+func (g *plastic) getCmCommand() string {
```

Comment:
This is a duplicate function, if we feel like we need this maybe it's better to add this logic to `runCommand` IF this is an issue for all executables (git is slow in this case so we fallback to the Windows variant)

Reply:
Oh. So this might not be needed at all? 
`cm.exe` (windows) and `cm` (others) can both be executed via `cm`. So maybe you'r right. I will test that.